CHANGES.md

```diff
@@ -60,15 +52,37 @@ Dev Improvements:
 - chore(dev) add theme picker to the tools/developer tool (#2770) [Josh Goebel][]
 - fix(dev) the Vue.js plugin no longer throws an exception when hljs is not in the global namespace [Kyle Brown][]
 
-Parser:
-
-- enh(grammars) allow `classNameAliases` for more complex grammars [Josh Goebel][]
-
 New themes:
 
 - *StackOverflow Dark* by [Jan Pilzer][]
 - *StackOverflow Light* by [Jan Pilzer][]
 
+
+Deprecations (*these will be mentioned in every release*):
+
+- HTML "merging" is deprecated. (#2873) [Josh Goebel][]
+  - HTML inside `<pre>` blocks will no longer be magically merged back into the
+  highlighted code's HTML result - it will instead be silently removed.
+  - Use a plugin if you truly need this functionality
```

Comment:
Is there a link to the plugin to make this more actionable?

Reply:
There is no plugin atm. Someone who really strongly needs the feature would need to contribute one.  Perhaps it should say "search for a plugin"?  I haven't even gotten around to making an issue thread for this one.  The existing code for that is not trivial for cutting and pasting... I also really don't want to encourage it's continued usage.  If a 3rd party author wants to bundle it into a plugin (the architecture changes here show this is possible) and then support it long-term, that would be fine. Otherwise I'd like to see the feature RIP.

It really never should have been added to core.

Reply:
https://github.com/highlightjs/highlight.js/issues/2889